src/NodeAutoCompleteViewExtension/NodeAutoCompleteViewExtension.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using Dynamo.Extensions;
+using Dynamo.Logging;
+using Dynamo.Wpf.Extensions;
+
+namespace Dynamo.NodeAutoComplete
+{
+    /// <summary>
+    /// This view extension tracks current clicked node port in Dynamo and 
+    /// try to suggest the next best node to connect that port.
+    /// </summary>
+    public class NodeAutoCompleteViewExtension : IViewExtension, ILogSource
+    {
+        private const String extensionName = "Node Auto Complete";
+
+        /// <summary>
+        /// Extension Name
+        /// </summary>
+        public string Name
+        {
+            get
+            {
+                return extensionName;
+            }
+        }
+
+        /// <summary>
+        /// GUID of the extension
+        /// </summary>
+        public string UniqueId
+        {
+            get
+            {
+                return "64F28473-0DCB-4E41-BB5B-A409FF6C90AD";
+            }
+        }
+
+        /// <summary>
+        /// Dispose function after extension is closed
+        /// </summary>
+        public void Dispose()
+        {
+            // Do nothing for now
+        }
+
+        public void Ready(ReadyParams readyParams)
+        {
+            // Do nothing for now
+        }
+
+        public void Shutdown()
+        {
+            this.Dispose();
```

Comment:
While we have been doing this, the life cycle of an extension will make it so that `Shutdown` is called and then `Dispose` is called as well. That results in `Dispose` being called two times, which can lead to exceptions. Not that it's causing issues now, because this extension is empty, but may lead to issues in the future so I guess we should avoid it.

Reply:
Hi @mmisol Can you point me where indicates Dispose() is going to be called twice?

Reply:
Took some time to dig it but found it. It's in `DynamoView.WindowClosed`. In that method you'll see `ext.Shutdown();` being called for each extension. Then `this.Dispose();` calls `viewExtensionManager.Dispose();` which will call `Dispose` on each extension.

Reply:
Ah, I see this is introduced as part of https://github.com/DynamoDS/Dynamo/pull/10672. Seems @mjkkirschner was doing it to make sure shutdown() and dispose() on extension level always get called. We have multiple ways to handle this then, we can either revert part of the Dispose() code from view extension manager or remove all instances  Dispose() calls on view extension shutdown.. Will ask team input tomorrow at standup